yt_dlp/extractor/itprotv.py

```diff
@@ -0,0 +1,174 @@
+            course_api = self._download_json(
+                f'https://api.itpro.tv/api/urza/v3/consumer-web/course?url={course_name}&brand=00002560-0000-3fa9-0000-1d61000035f3',
+                course_name, headers=headers, note='Fetching data from course API')
+
+        return course_api
+
+    def _get_episode_api_json(self, webpage, episode_id):
+        jwt = self._fetch_jwt(webpage)
+
+        if jwt:
+            headers = {'Authorization': 'Bearer ' + jwt}
+            episode_api = self._download_json(
+                f'https://api.itpro.tv/api/urza/v3/consumer-web/brand/00002560-0000-3fa9-0000-1d61000035f3/episode?url={episode_id}',
+                episode_id, headers=headers, note='Fetching data from episode API')
+
+        return episode_api
+
+    def _fetch_jwt(self, webpage):
+        return self._search_regex(r'{"passedToken":"([A-Za-z0-9-_]*\.[A-Za-z0-9-_]*\.[A-Za-z0-9-_]+)",', webpage, 'jwt')
+
+    def _check_if_logged_in(self, webpage):
+        if '{ member: null' in webpage:
+            raise self.raise_login_required(method='cookies')
+
+    def _real_extract(self, url):
+        QUALITIES = qualities(['low', 'medium', 'high', 'veryhigh'])
+
+        episode_id = self._match_id(url)
+        webpage = self._download_webpage(url, episode_id)
+        self._check_if_logged_in(webpage)
+
+        course_name = re.match(r'https?://.*/course/(?P<course_name>[0-9a-z-]+)/.*', url).group(1)
+        course_api = self._get_course_api_json(webpage, course_name)
+        course = course_api['course']
+
+        episode_api = self._get_episode_api_json(webpage, episode_id)
+        episode = episode_api['episode']
+
+        for i in range(len(course.get('topics'))):
+            if traverse_obj(course, ('topics', i, 'id'), expected_type=str) == episode.get('topic'):
+                chapter_name = traverse_obj(course, ('topics', i, 'title'), expected_type=str)
+                chapter_id = traverse_obj(course, ('topics', i, 'id'), expected_type=str)
+                chapter_number = i + 1
+
+        return {
+            'id': episode_id,
+            'title': episode['title'],
+            'description': episode.get('description'),
+            'thumbnail': episode.get('thumbnail'),
+            'formats': [
+                {'url': episode.get('jwVideo320Embed'), 'height': 320, 'quality': QUALITIES('low')},
+                {'url': episode.get('jwVideo480Embed'), 'height': 480, 'quality': QUALITIES('medium')},
+                {'url': episode.get('jwVideo720Embed'), 'height': 720, 'quality': QUALITIES('high')},
+                {'url': episode.get('jwVideo1080Embed'), 'height': 1080, 'quality': QUALITIES('veryhigh')}
+            ],
+            'duration': episode.get('length'),
+            'series': course.get('name'),
+            'series_id': course.get('url'),
+            'availability': self._availability(
+                needs_premium=False, needs_subscription=False, needs_auth=True,
```

Comment:
You said "most" videos need auth. Do all videos? Otherwise, this is wrong

Reply:
All videos need auth, most need a subscription

Reply:
Then why `needs_premium/needs_subscription=False`?

Reply:
I ended up removing this from the info_dict. Ultimately there is a mix of free and sub-only videos and from the APIs there is no way to tell between the two therefore as to not incorrectly assume I think it is best not to have it in the first place.